src/gourd_lib/config/mod.rs

```diff
@@ -380,18 +398,51 @@ impl Config {
     }
 }
 
-/// Is a value equal to its default value.
-/// Used for skipping serialisation,
-fn is_default<T: Default + PartialEq>(t: &T) -> bool {
-    t == &T::default()
-}
-
 /// Is the wrapper at its default value.
 /// Used for skipping serialisation.
 fn wrapper_is_default(w: &String) -> bool {
     w.eq(&WRAPPER_DEFAULT())
 }
 
+impl UserAfterscript {
+    /// Canonicalize the path to the afterscript executable.
+    pub fn canonicalize(&self, fs: &impl FileOperations) -> Result<Afterscript> {
+        let initial = match self {
+            Self::Complex(Afterscript { executable }) => executable,
+            Self::Simple(executable) => executable,
+        };
+        let executable = canon_path(initial, fs)?;
+        // on unix, check the file permissions and ensure the afterscript is executable.
+        #[cfg(unix)]
```

Comment:
No! Permissions can chage.

You should check when executing it, not when preparing the experiment.
The name of this function "`canonicalize`" is also very deceiving, it does so much more than canonicalize.

Reply:
permissions can change indeed, but is there really a scenario when they will *have* change between the experiment creation and status execution? because as a user I would rather when the experiment starts that there's no more debugging needed!

as for the name, agreed, will change.

Reply:
Your first point I agree with, and thus a check before execution is prudent, but the last point is personal preference. Why should the user care when afterscripts get executed?